examples/data-objects/vltava/src/fluidObjects/tabs/dataModel.ts

```diff
@@ -57,7 +55,11 @@ export class TabsDataModel extends EventEmitter implements ITabsDataModel {
     ) {
         super();
 
-        this.tabs = root.getSubDirectory("tab-ids");
+        const tabs = "tab-ids";
+        if (!root.hasSubDirectory(tabs)) {
+            root.createSubDirectory(tabs);
```

Comment:
This looks like it could theoretically cause multiple "createSubDirectory" ops to be generated, since it's outside of initializingFirstTime?  Does this actually need a subdirectory (looks like just operating on the root directory would be sufficient and sidestep the issue)?

Reply:
It will not, as it's called by data object in its initialization flow (there is no different between hasInitialized & initializingFirstTime re overall picture) which is guaranteed (by DataObject layer) to complete before data store is attached to file. I.e. other client will see it there.

That said, shouldn't this operation coalesce if two clients do it? I believe it should, if not, I think it's a bug.
As a general statement, I'd say this code is totally wrong - root map is not an API, we should stop passing it around and expect proper usage / composition of code. passing sub-directory would be slightly better over what we have today, but sub-directory does not have "valueChanged" event on it, I believe that's the reason this code needs to know about sub-directory in the first place. Would be great to keep improving this code in separate PRs :)